src/graphics/Screen.cpp

```diff
@@ -2150,13 +2151,23 @@ void Screen::setFrames(bool holdPosition = false)
 
     setFastFramerate(); // Draw ASAP
 
+    // In some situations, we'd like to return to the same frame
+    static size_t oldNumFrames = numframes;
```

Comment:
I am not sure of assigning oldNumFrames = numframes here. As we have already reset the value of numframes to 0 in row 2085, calculated new value in 2093 ? 
I think we just declare oldNumFrames. And assign value of numframes at the end in row 2169 , to use it next time ?

Reply:
It's maybe not the most common design pattern, but because it's `static`, it will only be declared there the first time that code runs. Every other time, it will preserve the value from the previous loop. It's something I've picked up lately because it feels like a convenient way to keep variables close to where they are used.

Reply:
Actually, I do see what you're getting at there. 

Just to be clear: because it's `static`, it will only be declared and initialized the very first time that method is called. Every other time, it will preserve the value from the end of the previous call. 

The very first time that method is called though, you're right, it will be initialized and then set again with the same value. In theory, the first setFrames call should always be `setFrames(false)`, so it would probably be okay to declare it with no initial value, and wait for it to be set at the end of that first call.

That being said, it is nice that it's being initialized before that `if(holdPosition)` block, just in case the very first call to setFrames is somehow `setFrames(true)`. There's probably no reason it couldn't be initialised with some fixed value like 2 instead. In theory that initial value will never be used; it'd just be insurance against some unexpected weirdness.

Honestly, I didn't put this much thought into it when I initially wrote that line. I think I probably moved it from somewhere else without realizing that it's a bit redundant now.

If you wanted to be really diligent, you could even initialise it as 0, and then `assert(oldNumFrames)` at the start of that `if(holdPosition)` block, to throw an error if it does somehow get user before it's meant to, although that might be overkill.

Whatever suits your sense of design!